cmd/root/root.go

```diff
@@ -126,5 +135,106 @@ func Execute(ctx context.Context, cmd *cobra.Command) error {
 		}
 	}
 
+	end := time.Now()
+	exitCode := 0
+	if err != nil {
+		exitCode = 1
+	}
+
+	uploadTelemetry(cmd.Context(), commandString(cmd), start, end, exitCode)
 	return err
 }
+
+// We want child telemetry processes to inherit environment variables like $HOME or $HTTPS_PROXY
```

Comment:
Did you consider allowlist of env vars here instead? Might be simpler to do than with filtering out.

Reply:
I did. It's not a good fit here because we will never get a signal that some important environment variable is not being passed to the telemetry worker because the telemetry upload itself is invisible.

We do use an allow list for env vars when we run `terraform apply`. It's fine there because the command is in the happy path of the deployment and thus we get bug reports for example when `HTTP_PROXY` or some authentication env var for Azure is not passed.

Reply:
> some important environment variable is not being passed to the telemetry worker 

What are those variables, can you give some examples?

BTW, did you check what approach we use when launching terraform? allowlist of blocklist?

Reply:
> What are those variables, can you give some examples?

Here's a few examples: https://github.com/databricks/cli/blob/9056fb58b78ff9eb3d65ecaa905ea6205af3aca3/bundle/deploy/terraform/init.go#L91

For instance, we recognized `AZURE_CONFIG_DIR` was required because of a bug report.

> BTW, did you check what approach we use when launching terraform? allowlist of blocklist?

Yeah, terraform uses an allowlist.